emirge.py

```diff
@@ -830,45 +835,36 @@ def cluster_sequences2(self, fastafilename):
         tocleanup.append(tmp_fastafilename)
         tmp_fastafile = pysam.Fastafile(tmp_fastafilename)
         tocleanup.append("%s.fai"%(tmp_fastafilename))
-        # do global alignments with USEARCH/UCLUST.
-        # I don't use --cluster because it doesn't report alignments
-        # usearch is fast but will sometimes miss things -- I've tried to tune params as best as I can.
+        # modified to align using vsearch
         # Also, I use a lower %ID thresh than specified for joining because I really calculate %ID over *mapped* sequence positions.
 
-        sens_string = "--maxaccepts 8 --maxrejects 256"
-        uclust_id = 0.80
-        algorithm="-usearch_global"
-        # uclust_id = self.cluster_thresh - 0.05
+        sens_list = ["--maxaccepts", "8", "--maxrejects", "256"]
 
-        # if em.iteration_i > 10:
-        # num_seqs = len([x for x in self.probN if x is not None])
         assert num_seqs == len([x for x in self.probN if x is not None])
         if num_seqs < 1000:
-            sens_string = "--maxaccepts 16 --maxrejects 256"
+            sens_list = ["--maxaccepts", "16", "--maxrejects", "256"]
         if num_seqs < 500:
-            sens_string = "--maxaccepts 32 --maxrejects 256"
-        if num_seqs < 150:
-            algorithm="-search_global"
-            sens_string = "--maxaccepts 0 --maxrejects 0"  # slower, but more sensitive.
-        # if really few seqs, then no use not doing smith-waterman or needleman wunsch alignments
-        if num_seqs < 50:
-            algorithm="-search_global"
-            sens_string = "-fulldp"
-
-        # there is a bug in usearch threads that I can't figure out (fails with many threads).  Currently limiting to max 6 threads
-        usearch_threads = min(6, self.n_cpus)
-        cmd = "usearch %s %s --db %s --id %.3f -quicksort -query_cov 0.5 -target_cov 0.5 -strand plus --userout %s.us.txt --userfields query+target+id+caln+qlo+qhi+tlo+thi -threads %d %s"%\
-              (algorithm,
-               tmp_fastafilename, tmp_fastafilename,
-               uclust_id,
-               tmp_fastafilename,
-               usearch_threads,
-               sens_string)
+            sens_list = ["--maxaccepts", "32", "--maxrejects", "256"]
+        if num_seqs < 150: 
+            sens_list = ["--maxaccepts", "0", "--maxrejects", "0"]  # slower, but more sensitive.
+
+        cmd = [
+                'vsearch',
+                '--threads', str(self.n_cpus),
+                '--allpairs_global',
+                tmp_fastafilename,
+                '--acceptall',
+                '-strand', 'plus',
+                '--userout', "{}.us.txt".format(tmp_fastafilename),
+                '--userfields', 'query+target+id+caln+qlo+qhi+tlo+thi', 
+            ] + sens_list
+        result = subprocess.call(
+            args=cmd
+        )
```

Comment:
You need to either keep using `check_call` which raises an exception if `vsearch` fails or add error handling based on the result code.

Reply:
Glad to convert to check call